### ISSUE TITLE
panic: interface conversion: source.ResourceClient is *main.client, not *source.clientWrapper

### ISSUE THREAD

**Issue by embroede** [dragonflyoss/Dragonfly2#1757]
### Bug report:

We are seeing the below in /var/log/dragonfly/daemon/core.log when attempting to use a custom resource plugin (out-of-tree) compiled into the dfdaemon image.

The issue seems to be coming from line 392 of source_client.go, attempting to use clientWrapper though the ResourceClient is client.

I do notice this function has changed in main. Maybe that was to fix this issue?

```
panic: interface conversion: source.ResourceClient is *main.client, not *source.clientWrapper

goroutine 121 [running]:
d7y.io/dragonfly/v2/pkg/source.GetMetadata(0xc00026dc00)
        /go/src/d7y.io/dragonfly/v2/pkg/source/source_client.go:392 +0x1b0
d7y.io/dragonfly/v2/client/daemon/peer.(*pieceManager).DownloadSource(0xc00094fdd0, {0x2046370, 0xc000939260}, {0x204d788?, 0xc0007d62c0?}, 0xc000b2e180, 0x0)
        /go/src/d7y.io/dragonfly/v2/client/daemon/peer/piece_manager.go:299 +0x2b3
d7y.io/dragonfly/v2/client/daemon/peer.(*peerTaskConductor).backSource(0xc0007d62c0)
        /go/src/d7y.io/dragonfly/v2/client/daemon/peer/peertask_conductor.go:498 +0xe5
d7y.io/dragonfly/v2/client/daemon/peer.(*peerTaskConductor).waitFirstPeerPacket(0xc0007d62c0)
        /go/src/d7y.io/dragonfly/v2/client/daemon/peer/peertask_conductor.go:1012 +0x3d6
d7y.io/dragonfly/v2/client/daemon/peer.(*peerTaskConductor).pullPiecesFromPeers(0xc0007d62c0, 0xc000245440?)
        /go/src/d7y.io/dragonfly/v2/client/daemon/peer/peertask_conductor.go:866 +0x38
created by d7y.io/dragonfly/v2/client/daemon/peer.(*peerTaskConductor).pullPiecesWithP2P
        /go/src/d7y.io/dragonfly/v2/client/daemon/peer/peertask_conductor.go:550 +0x172

```

### Expected behavior:

The download to happen

### How to reproduce it:

Attempt to download a resource with a custom resource plugin.

### Environment:
- Dragonfly version: v2.0.5
- OS: Linux
- Kernel (e.g. `uname -a`): 4
- Others:

**Comment by jim3ma** [comment]
Fixed in #1724, diff: https://github.com/dragonflyoss/Dragonfly2/pull/1724/files#diff-6adaf9ce78a62caa6950bdbff407ec787dfe31ef1fec0e37c4c927071828c8c6L392

**Comment by embroede** [comment]
Ok great, we will give that a try, thanks!
